relay/nakama/receipt/notifications_test.go

```diff
@@ -0,0 +1,122 @@
+package receipt
+
+import (
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"pkg.world.dev/world-engine/relay/nakama/mocks"
+	"pkg.world.dev/world-engine/relay/nakama/testutils"
+	"strings"
+	"testing"
+	"time"
+)
+
+// Test that the Notifications system works as expected with the Dispatcher and a Mock Server
+func TestNotifierIntegrationWithDispatcher(t *testing.T) {
+	nk := mocks.NewNakamaModule(t)
+	logger := &testutils.NoOpLogger{}
+
+	mockServer := setupMockServer()
+	defer mockServer.Close()
+
+	rd := NewReceiptsDispatcher()
+	notifier := NewNotifier(logger, nk, rd)
+
+	txHash := "hash1"
+	userID := "user456"
+	notifier.AddTxHashToPendingNotifications(txHash, userID)
+	expectedData := map[string]any{
+		"txHash": txHash,
+		"result": map[string]any{"status": "success"},
+		"errors": []string{},
+	}
+	nk.On("NotificationSend",
+		mock.Anything, userID, mock.Anything, expectedData, mock.Anything, mock.Anything, mock.Anything,
+	).Return(nil).Once()
+
+	go rd.Dispatch(logger)
+	go rd.PollReceipts(logger, strings.TrimPrefix(mockServer.URL, "http://"))
+
+	time.Sleep(time.Second)
+
+	nk.AssertExpectations(t)
+}
+
+func TestAddTxHashToPendingNotifications(t *testing.T) {
+	logger := &testutils.NoOpLogger{}
+	nk := mocks.NewNakamaModule(t)
+	rd := NewReceiptsDispatcher()
+	notifier := NewNotifier(logger, nk, rd)
+
+	txHash := "hash1"
+	userID := "user456"
+
+	notifier.AddTxHashToPendingNotifications(txHash, userID)
+
+	info, exists := notifier.txHashToTargetInfo[txHash]
+	assert.True(t, exists, "TxHash should exist in the map after being added")
+	assert.Equal(t, userID, info.userID, "UserID associated with TxHash does not match")
+}
+
+func TestHandleReceipt(t *testing.T) {
+	logger := &testutils.NoOpLogger{}
+	nk := mocks.NewNakamaModule(t)
+	rd := NewReceiptsDispatcher()
+	notifier := NewNotifier(logger, nk, rd)
+
+	txHash := "hash1"
+	userID := "user456"
+
+	// Assert that "NotificationSend" is called with the given params
+	nk.On("NotificationSend",
+		mock.Anything, userID, "subject",
+		mock.AnythingOfType("map[string]interface {}"), 1, "", false).Run(func(args mock.Arguments) {
+		argData := args.Get(3).(map[string]any) //nolint:errcheck // [not important]
+		assert.Nil(t, argData["result"])
+		assert.Nil(t, argData["errors"])
+	}).Return(nil).Once()
+
+	notifier.txHashToTargetInfo[txHash] = targetInfo{
+		createdAt: time.Now(),
+		userID:    userID,
+	}
+
+	receipt := &Receipt{TxHash: txHash}
+	err := notifier.handleReceipt(receipt)
+	assert.NoError(t, err, "Handling receipt should not error")
+
+	_, exists := notifier.txHashToTargetInfo[txHash]
+	assert.False(t, exists, "TxHash should be removed from map after processing")
+
+	nk.AssertExpectations(t)
```

Comment:
Yeah, I left this in because when I first came across these Mocks I was confused where/how the Mock requirements were being executed. I can remove tho